src/System.Private.CoreLib/src/System/Threading/LazyInitializer.cs

```diff
@@ -169,6 +156,33 @@ public static T EnsureInitialized<T>(ref T target, ref bool initialized, ref obj
         }
 
         /// <summary>
+        /// Ensure the lock object is intialized.
+        /// </summary>
+        /// <param name="syncLock">A reference to a location containing a mutual exclusive lock. If <paramref name="syncLock"/> is null,
+        /// a new object will be instantiated.</param>
+        /// <returns>Initialized lock object.</returns>
+        private static object EnsureLockInitialized(ref object syncLock) =>
+            syncLock ??
+            Interlocked.CompareExchange(ref syncLock, new object(), null) ??
+            syncLock;
+
+        /// <summary>
+        /// Initializes a target reference type with a specified function if it has not already been initialized.
+        /// </summary>
+        /// <typeparam name="T">The type of the reference to be initialized. Has to be reference type.</typeparam>
+        /// <param name="target">A reference of type <typeparamref name="T"/> to initialize if it has not already been initialized.</param>
+        /// <param name="syncLock">A reference to an object used as the mutually exclusive lock for initializing
+        /// <paramref name="target"/>. If <paramref name="syncLock"/> is null, a new object will be instantiated.</param>
+        /// <param name="valueFactory">The <see cref="T:System.Func{T}"/> invoked to initialize the reference.</param>
+        /// <returns>The initialized value of type <typeparamref name="T"/>.</returns>
+        public static T EnsureInitialized<T>(ref T target, ref object syncLock, Func<T> valueFactory) where T : class =>
+            // According to the abstract ECMA CLI memory model, this read should be volatile, due to possible reorderings with
+            // subsequent reads from fields of "target."  However, as of today (7/12/12), Volatile.Read<T> is quite slow, and in practice
+            // this read does *not* need to be volatile, because our memory model preserves ordering of data-dependent reads.
+            // Another occurrence at line 86.
+            target ?? EnsureInitializedCore<T>(ref target, ref syncLock, valueFactory);
```

Comment:
I have changed Volatile.Read to normal read to match that at line 86. Please check the comment at line 182 and 85. Will remove those comments if inappropriate. Kindly advise.